File: src/main/java/com/fish/extendedae_plus_client/widgets/button/EAEPActionButton.java
package com.fish.extendedae_plus_client.widgets.button;

import java.util.function.Consumer;

public class EAEPActionButton extends EAEPButton {
    protected final EAEPActionItems action;

    public EAEPActionButton(EAEPActionItems action, Consumer<EAEPActionItems> onPress) {
        super(button -> onPress.accept(button.getAction()));
        this.action = action;
        this.updateTooltip();
    }

    @Override
    public EAEPActionItems getAction() {
        return this.action;
    }
}

File: src/main/kotlin/com/fish/extendedae_plus_client/render/widgets/button/EAEPActionButton.kt
package com.fish.extendedae_plus_client.render.widgets.button

import java.util.function.Consumer

open class EAEPActionButton(override val action: EAEPActionItems, onPress: Consumer<EAEPActionItems?>) : EAEPButton(
    { button: EAEPButton -> onPress.accept(button.action) }
) {
    init {
        this.updateTooltip()
    }
}
